backup: TestCloudBackupRestoreAzure failed

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/18844743?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/18844743?buildTab=artifacts#/) on master @ [34e34c2d5b9598441b935f03a794277e9329a256](https://github.com/cockroachdb/cockroach/commits/34e34c2d5b9598441b935f03a794277e9329a256):


```
=== RUN   TestCloudBackupRestoreAzure
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a618b6da2371c05e85e1823e891dd1ac/logTestCloudBackupRestoreAzure47151802
    test_log_scope.go:76: use -show-logs to present logs inline
    backup_cloud_test.go:268: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a618b6da2371c05e85e1823e891dd1ac/logTestCloudBackupRestoreAzure47151802
--- FAIL: TestCloudBackupRestoreAzure (252.93s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_explicit
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:945: scanned job rows did not contain a backup!
    testutils.go:313: no Invalid Descriptors
    --- FAIL: TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_explicit (117.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47897

--- a/pkg/backup/backup_test.go
+++ b/pkg/backup/backup_test.go
@@ -912,7 +912,7 @@ func backupAndRestore(
 
 		found := false
 		stmt := `
-SELECT payload FROM "".crdb_internal.system_jobs ORDER BY created DESC LIMIT 10
+SELECT payload FROM "".crdb_internal.system_jobs WHERE job_type = 'BACKUP' ORDER BY created DESC LIMIT 10
 `
 		rows := sqlDB.Query(t, stmt)
 		for rows.Next() {

